### DISCUSSION TITLE
OpenFeature

### DISCUSSION THREAD

**Discussion by sagikazarmark** [flipt-io/flipt#727] [General]
I'm not very familiar with the project yet, but OpenFeature is the newest thing in the feature flag ecosystem: https://open-feature.github.io/

Might worth exploring if it makes sense for flipt.

**Comment by markphelps** [comment]
Thanks @sagikazarmark !  I just signed up to their newsletter to stay up to date. I hadn't heard of this project

**Comment by sagikazarmark** [comment]
Looks like there is an official provider for OpenFeature now: https://github.com/flipt-io/flipt-openfeature-provider-go

Very cool! :tada: :tada: :tada: :tada: :tada:

I was wondering though if using the provider (and this the OpenFeature SDKs) there is anything that's unavailable. If so, it would be nice to have a comparison between the OpenFeature API and the "native" Flipt SDKs.

WDYT @markphelps ?

**Comment by markphelps** [comment]
Hey @sagikazarmark ! Thank you for bringing OpenFeature to our attention in the first place! It's a great project!

> I was wondering though if using the provider (and this the OpenFeature SDKs) there is anything that's unavailable. 

Do you mean anything that the OpenFeature SDK exposes that Flipt does not? Or vice-versa?

If the latter, there is definitely a lot that Flipt exposes that the OpenFeature SDK does not, mostly around CRUD or resources + auth + other 'management' type things. I can definitely see the value in documenting all this though!

AFAIK OpenFeature is still mainly concerned with 'evaluation' so the main overlap with OpenFeature (and the Flipt provider) and the 'native' Flipt SDKs is in the `GetFlag` and `Evaluate` methods currently

**Reply by sagikazarmark** [reply]
> Do you mean anything that the OpenFeature SDK exposes that Flipt does not? Or vice-versa?

Well, I imagine that Flipt has a richer API, so I guess vica-versa, but if there is anything that OpenFeature exposes and Flipt does not, it could go into a comparison.

I think it would be useful to have a document that helps users decide whether they want to go with the OpenFeature SDK or with Flipt directly. I guess the most important part is evaluation. From what I can tell, OpenFeature does not have a batch evaluation API for example.

**Reply by markphelps** [reply]
👍🏻 Makes sense. Will take a crack at documenting the differences tomorrow
